windows-agent/internal/distros/database/database_test.go

```diff
@@ -497,7 +501,16 @@ func TestDatabaseCleanup(t *testing.T) {
 
 			databaseFromTemplate(t, dbDir, distros...)
 
-			db, err := database.New(ctx, dbDir)
+			var cleanupCalled atomic.Bool
+			var cleanupFunc func(string)
+			if tc.cleanupFunc {
+				cleanupFunc = func(d string) {
+					require.True(t, strings.EqualFold(tc.markDistroUnreachable, d), "Unexpected cleaned up distro name")
+					cleanupCalled.Store(true)
+				}
+			}
```

Comment:
Using require in this case is discouraged because it may panic in a goroutine other than the test main (this callback will be executed in another goroutine for sure).

```suggestion
			if tc.cleanupFunc {
				cleanupFunc = func(d string) {
					cleanupCalled.Store(strings.EqualFold(tc.markDistroUnreachable, d))
				}
			}
```